openapi.yaml

```diff
@@ -11717,6 +11717,135 @@ paths:
         - lang: CLI
           source: >
             linode-cli object-storage cancel
+  /object-storage/buckets/{clusterId}/{bucket}/ssl:
+    x-linode-cli-command: object-storage
+    parameters:
+    - name: clusterId
+      in: path
+      description: The ID of the cluster this bucket exists in.
+      required: true
+      schema:
+        type: string
+    - name: bucket
+      in: path
+      description: The bucket name.
+      required: true
+      schema:
+        type: string
+    get:
+      operationId: getObjectStorageSSL
+      x-linode-cli-action: ssl-view
+      servers:
+      - url: https://api.linode.com/v4
+      summary: View Object Storage SSL Cert
+      description: |
+        Returns whether this bucket has a corresponding SSL certificate that was
+        uploaded by a user.
+      tags:
+      - Object Storage
+      security:
+      - personalAccessToken: []
+      - oauth:
+        - object_storage:read_only
+      responses:
+        '200':
+          description: >
+            Returns whether this bucket has a corresponding SSL certificate
+            that was uploaded by a user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ObjectStorageSSLResponse'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      x-code-samples:
+      - lang: Shell
+        source: >
+          curl -H "Authorization: Bearer $TOKEN" \
+            https://api.linode.com/v4/object-storage/buckets/us-east-1/example-bucket/ssl
+      - lang: CLI
+        source: >
+          linode-cli object-storage ssl-view \
+            us-east-1 example-bucket
+    post:
+      operationId: createObjectStorageSSL
+      x-linode-cli-action: ssl-upload
+      servers:
+      - url: https://api.linode.com/v4
+      summary: Upload Object Storage SSL Cert
+      description: |
+        Upload an SSL certificate to be served when you visit your Object Storage bucket via HTTPS.
+        Your SSL certificate and private key are stored encrypted at rest.
+      tags:
+      - Object Storage
+      security:
+      - personalAccessToken: []
+      - oauth:
+        - object_storage:read_write
+      requestBody:
+        description: Upload this SSL certificate with its corresponding secret key.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ObjectStorageSSLResponse'
+      responses:
+        '200':
+          description: Returns whether this bucket has a corresponding SSL certificate that was uploaded by a user.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ObjectStorageSSLResponse'
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+      x-code-samples:
+      - lang: Shell
+        source: >
```

Comment:
I've just put up a PR to change this a bit. It's kind of a pain to have to strip newlines to fit PEM certs within a json string, so instead we'll be accepting the certificate and key as file inputs, similar to how we accept support ticket attachments.

Reply:
@adammohammed I added some info about being able to also upload a .crt file. If this feature doesn't get added, I'll revert the commit.

Reply:
@leslitagordita Yeah looks like if we add this we'll have to do the same for node balancers. So I think let's revert it for now. Those changes will have to come later.